packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectHelper.kt

```diff
@@ -98,28 +296,14 @@ internal object RealmObjectHelper {
         // In order to be able to use Kotlin's nullability handling baked into the accessor we need
         // to ask Core for the current value and return null if the value itself is null, returning
         // an instance of the wrapper otherwise - not optimal but feels quite idiomatic.
-        val currentValue = RealmInterop.realm_get_value(obj.objectPointer, propertyKey)
-        return when {
-            currentValue.value != null -> ManagedMutableRealmInt(obj, propertyKey, converter)
-            else -> null
+        return getterScope {
+            when (realm_get_value(obj.objectPointer, propertyKey, allocRealmValueT())) {
+                null -> null
```

Comment:
Don't really get why this is not baked into the allocator, but since we are now abandoning generic code paths (😞) I guess it doesn't matter.